### PR TITLE
[CDAP-18124] Fix null label added without reuse enabled

### PR DESCRIPTION
**Pull request by tivv**

### CODE REVIEW

**Review by dli357 — APPROVED** [review]

File: cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/provisioner/Cluster.java [hunk 1]
```diff
@@ -33,7 +34,8 @@
   public Cluster(String name, ClusterStatus status, Collection<Node> nodes, Map<String, String> properties) {
     this.name = name;
     this.status = status;
-    this.nodes = Collections.unmodifiableCollection(nodes);
+    this.nodes = nodes instanceof List ? Collections.unmodifiableList((List<Node>) nodes) :
```
Comment (chtyim) [inline cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/provisioner/Cluster.java]: Why this change is needed? Also, shouldn't it better to do a defensive copy?

Comment (tivv) [reply cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/provisioner/Cluster.java]: It's because equals do not work with Collections.unmodifiableCollection :(. It affects testing, but also anyone else using equals. I can do a deep copy
